app/src/main/java/com/github/wnder/HistoryAdapter.java

```diff
@@ -80,6 +90,34 @@ public int getItemCount() {
         return pictureList.size();
     }
 
+    private void showPopup(ViewHolder holder, Bitmap bmp, String pictureID){
+        PopupMenu popup = new PopupMenu(holder.itemView.getContext(), holder.itemView);
+        popup.setOnMenuItemClickListener(item -> {
+            int id = item.getItemId();
+            if(id == R.id.save){
+                saveToGallery(pictureID, bmp, holder);
+                return true;
+            }
+            return false;
+        });
+        MenuInflater inflater = popup.getMenuInflater();
+        inflater.inflate(R.menu.history_menu, popup.getMenu());
+        popup.show();
+    }
+
+    private void saveToGallery(String pictureID, Bitmap bmp, ViewHolder holder){
+        if(pictureID.equals(Picture.UNINITIALIZED_ID)){
+            //Snack bar
+            Snackbar snackbar = Snackbar.make(holder.itemView, R.string.bar_save_is_impossible, BaseTransientBottomBar.LENGTH_SHORT);
+            snackbar.show();
+        }
+        else{
+            MediaStore.Images.Media.insertImage(holder.itemView.getContext().getContentResolver(), bmp, pictureID, "");
+            Snackbar snackbar = Snackbar.make(holder.itemView, R.string.bar_save_is_ok, BaseTransientBottomBar.LENGTH_SHORT);
+            snackbar.show();
```

Comment:
You can display snackbars without a variable: 
```
Snackbar.make(holder.itemView, R.string.bar_save_is_ok, Snackbar.LENGTH_SHORT)
    .show();
```

Reply:
It's done.